packages/pytest-simcore/src/pytest_simcore/helpers/catalog_rpc_server.py

```diff
@@ -25,15 +25,17 @@
 )
 from models_library.services_types import ServiceKey, ServiceVersion
 from models_library.users import UserID
-from pydantic import NonNegativeInt, TypeAdapter
+from pydantic import NonNegativeInt, TypeAdapter, validate_call
+from pytest_mock import MockType
 from servicelib.rabbitmq._client_rpc import RabbitMQRPCClient
 
 
 class CatalogRpcSideEffects:
     # pylint: disable=no-self-use
+    @validate_call(config={"arbitrary_types_allowed": True})
     async def list_services_paginated(
         self,
-        rpc_client: RabbitMQRPCClient,
+        rpc_client: RabbitMQRPCClient | MockType,
```

Comment:
Strange, I would try and drop the MockType and use something like: 
```python
AsyncMock(spec=RabbitMQRPCClient)
```
please check if this works in this situation

Reply:
aha good input! thx

Reply:
I cannot use this because I need a type. For a moment I thought there was some meta-class as `AsyncMock[RabbitMQRPCClient]`

Reply:
Note that these annotations are used to validate i/o (see the `@validate_call`)